specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/inference.json

```diff
@@ -0,0 +1,1443 @@
+          "enum": [
+            "SystemAssigned",
+            "SystemAssigned,UserAssigned",
+            "UserAssigned",
+            "None"
+          ],
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "$ref": "#/definitions/UserAssignedIdentities",
+          "description": "The user assigned identities associated with the resource."
+        }
+      },
+      "description": "Identity for the resource."
+    },
+    "UserAssignedIdentities": {
+      "description": "dictionary containing all the user assigned identities, with resourceId of the UAI as key.",
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/UserAssignedIdentity"
+      }
+    },
+    "UserAssignedIdentity": {
+      "description": "User Assigned Identity",
+      "properties": {
+        "principalId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The principal ID of the user assigned identity."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant ID of the user assigned identity."
+        },
+        "clientId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The clientId(aka appId) of the user assigned identity."
+        }
+      }
+    },
+    "PaginatedInferenceEndpointsList": {
+      "type": "object",
+      "description": "Paginated list of Machine Learning Inference Endpoint objects wrapped in ARM resource envelope.",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "description": "An array of Machine Learning Inference Endpoint objects wrapped in ARM resource envelope.",
+          "items": {
+            "$ref": "#/definitions/InferenceEndpointResource"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A continuation link (absolute URI) to the next page of results in the list."
+        }
+      }
+    },
+    "PaginatedInferenceEndpointDeploymentsList": {
+      "type": "object",
+      "description": "Paginated list of Machine Learning Inference Deployment objects wrapped in ARM resource envelope.",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "description": "An array of Machine Learning Inference Deployment  objects wrapped in ARM resource envelope.",
+          "items": {
+            "$ref": "#/definitions/InferenceEndpointDeploymentResource"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A continuation link (absolute URI) to the next page of results in the list."
+        }
+      }
+    },
+    "SystemData": {
```

Comment:
>SystemData [](start = 5, length = 10)

Instead of defining these objects in your file please refer to [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L45](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L45)